Added Hyprspace Overview (AGS Alternive)

0xl30 opened the pull request:
# Pull Request

## Description

Please read these instructions and remove unnecessary text.

- Try to include a summary of the changes and which issue is fixed.
- Also include relevant motivation and context (if applicable).
- List any dependencies that are required for this change. (e.g., packages or other PRs)
- Provide a link if there is an issue related to this pull request. e.g., Fixes # (issue)
- Please add Reviewers, Assignees, Labels, Projects, and Milestones to the PR. (if applicable)

## Type of change

Please put an `x` in the boxes that apply:

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] **Documentation update** (non-breaking change; modified files are limited to the documentations)
- [ ] **Technical debt** (a code change that does not fix a bug or add a feature but makes something clearer for devs)
- [ x] **Other** (provide details below)

## Checklist

Please put an `x` in the boxes that apply:

- [ x] I have read the [CONTRIBUTING](https://github.com/JaKooLit/Hyprland-Dots/blob/main/CONTRIBUTING.md) document.
- [x ] My code follows the code style of this project.
- [ x] My commit message follows the [commit guidelines](https://github.com/JaKooLit/Hyprland-Dots/blob/main/CONTRIBUTING.md#git-commit-messages).
- [ ] My change requires a change to the documentation.
- [ ] I want to add something in Hyprland-Dots wiki.
- [ ] I have added tests to cover my changes.
- [ x] I have tested my code locally and it works as expected.
- [ x] All new and existing tests passed.

## Screenshots
![7mb](https://github.com/user-attachments/assets/c0151e31-504e-434f-bc8d-8397fa771fb9)


(if appropriate)

## Additional context
Need install hyprland hyprspace plugins
```
hyprpm add https://github.com/KZDKM/Hyprspace
hyprpm enable Hyprspace 
```
Add any other context about the problem here.

dwilliam62 commented:
Thank you for submitting this. 

My recommendation is that this not get added to the project.   User's are free to add them on their own.   Typically with every Hyprland Update you end up having to deal with updating the plugins or get kernel mismatch errors. 

  If the project provides plugins that will put the burden back on the project staff to support them.   Including the documentation  on how to install, use, update and remove the plugin.  Plus having to test this on all the supported distros.   
  
Finally, NIXOS manages plugins differently from Arch, etc...    So that's an additional requirement  to support.

dwilliam62 commented:
Plugins also require additional dependencies   
Could not clone the plugin repository. Dependencies not satisfied. Hyprpm requires: cmake, meson, cpio, pkg-config

dwilliam62 commented:
I am curious what your keybinding to toggle this is.  Please post it   thanks

0xl30 commented:
> I am curious what your keybinding to toggle this is. Please post it thanks

Thank you for your comment.
I don’t use NixOS and am not very familiar with it. The developer will verify everything before merging, and if everything is fine will the code be added to the project.
I have configured this only for my own setup, so I can only provide the idea(not full code). The rest of the responsibility to finalize everything lies with the developer.

dwilliam62 commented:
> > I am curious what your keybinding to toggle this is. Please post it thanks
> 
> Thank you for your comment. I don’t use NixOS and am not very familiar with it. The developer will verify everything before merging, and if everything is fine will the code be added to the project. I have configured this only for my own setup, so I can only provide the idea(not full code). The rest of the responsibility to finalize everything lies with the developer.

I realize this.   What are you using for a keybind to activate this ?

0xl30 commented:
> > > I am curious what your keybinding to toggle this is. Please post it thanks
> > 
> > 
> > Thank you for your comment. I don’t use NixOS and am not very familiar with it. The developer will verify everything before merging, and if everything is fine will the code be added to the project. I have configured this only for my own setup, so I can only provide the idea(not full code). The rest of the responsibility to finalize everything lies with the developer.
> 
> I realize this. What are you using for a keybind to activate this ?

bind = $mainMod, A, exec, hyprctl dispatch overview:toggle #hyprspace Overview

JaKooLit commented:
Thanks for this....

let me think about it and find out how to implement / automate in installation

JaKooLit commented:
Thanks bro for the extensive test

I wont merge it then

Sounds like a nightmare

@0xl30  Sorry bro. But thanks alot..

0xl30 commented:
> I've been experimenting with this plugin. Worked fine. Updated Hyprland. Updated plugins. `hyprm update` Restarted Hyprland (even rebooted) PC The plugin no longer works. "Invalid dispatcher" I also added the now optional hyprland-protocols for plugin support. these kinds of issues are a repeat of my previous experiences with hyprland plugins. I am going to recommend that he not add in plugins to the default config. it becomes a support nightmare otherwise. Users are free to add this in on their own.

it's unofficial plugins.
Whenever Hyprland changes something, the plugin breaks.
I mentioned before that this is an alternative to AGS (there are many alternative implementations(like waybar temp code py and bash) and I created this PR based on those).
Users can choose to use it if they want(i don't recommend too add this by default you can add it alternative)--it's much more advanced and visually appealing.

dwilliam62 commented:
> it's unofficial plugins. Whenever Hyprland changes something, the plugin breaks. I mentioned before that this is an alternative to AGS (there are many alternative implementations(like waybar temp code py and bash) and I created this PR based on those). Users can choose to use it if they want(i don't recommend too add this by default you can add it alternative)--it's much more advanced and visually appealing.

 Understood.  I had hoped the plugins had gotten better with time.  I've had issues even with official plugins.   FYI:  When someone creates a PR we take that as they are suggesting it as a basic feature of the configuration.  

 Thanks again
